sonic-xcvrd/scripts/xcvrd

```diff
@@ -373,82 +499,37 @@ def main():
         log_error("Error reading port info (%s)" % str(e), True)
         sys.exit(2)
 
-    # Connect to STATE_DB and create transceiver info/dom info table
-    state_db = swsscommon.DBConnector(swsscommon.STATE_DB,
-                                     REDIS_HOSTNAME,
-                                     REDIS_PORT,
-                                     REDIS_TIMEOUT_MSECS)
-    int_tbl = swsscommon.Table(state_db, "TRANSCEIVER_INFO")
-    dom_tbl = swsscommon.Table(state_db, "TRANSCEIVER_DOM_SENSOR")
+    # Make sure this daemon started after all port configured
+    log_info("Wait for port config is done")
+    wait_for_port_config_done()
 
-    # Connect to APPL_DB abd subscribe to PORT table notifications
-    appl_db = swsscommon.DBConnector(swsscommon.APPL_DB,
-                                     REDIS_HOSTNAME,
-                                     REDIS_PORT,
-                                     REDIS_TIMEOUT_MSECS)
+    # Post all the current interface dom/sfp info to STATE_DB
+    log_info("Post all port DOM/SFP info to DB")
+    post_port_sfp_dom_info_to_db()
 
-    sel = swsscommon.Select()
-    sst = swsscommon.SubscriberStateTable(appl_db, swsscommon.APP_PORT_TABLE_NAME)
-    sel.addSelectable(sst)
-
-    # Make sure this daemon started after all port configured.
-    while True:
-        (state, c) = sel.select(SELECT_TIMEOUT_MSECS)
-        if state == swsscommon.Select.TIMEOUT:
-            continue
-        if state != swsscommon.Select.OBJECT:
-            log_warning("sel.select() did not return swsscommon.Select.OBJECT")
-            continue
+    # Start the dom sensor info update thread
+    dom_info_update = dom_info_update_task()
+    dom_info_update.task_run()
 
-        (key, op, fvp) = sst.pop()
-        if key in ["PortConfigDone", "PortInitDone"]:
-            break
+    # Start the sfp state info update process
+    sfp_state_update = sfp_state_update_task()
+    sfp_state_update.task_run()
 
-    # Post all the current interface SFP info to STATE_DB
-    logical_port_list = platform_sfputil.logical
-    for logical_port_name in logical_port_list:
-        post_port_sfp_info_to_db(logical_port_name, int_tbl)
-        post_port_dom_info_to_db(logical_port_name, dom_tbl)
+    # Start main loop
+    log_info("Start daemon main loop")
 
-    # Start the dom sensor info update timer thread
-    dom_info_update = dom_info_update_task(dom_tbl)
-    dom_info_update.task_run()
+    while RUN:
+        time.sleep(TIMEOUT_SECS)
```

Comment:
main task is idling, original design using the main task to listen to the sfp change event, can we reduce one thread here? have you compared the time consuming of xvrd before and after, any significant increase?

Reply:
@keboliu before changes on stop ~10 sec, after changes ~1 sec.

The idea is to use main daemon process to:
1. Do the init
2. Handle signals
2. Manage tasks

This is a different architecture approach.